### PR TITLE
Do not create companions that will be dropped later.

### PR DESCRIPTION
**Pull request by DarkDimius**
Fix blocker bug reported in #1114
I dislike this fix as now phase needs to know in advance
if it will ever need a companion for the class.
On the bright side, this change makes it clear
which phases need companions

### CODE REVIEW

**Comment by DarkDimius** [comment]
@odersky please review.

**Comment by DarkDimius** [comment]
Single test failed: TASTY pickling with [error] java.lang.OutOfMemoryError: GC overhead limit exceeded

Fix passes all the test on my machine and is ready for review.

File: src/dotty/tools/dotc/transform/LazyVals.scala [hunk 1]
```diff
@@ -46,6 +47,12 @@ class LazyVals extends MiniPhaseTransform with IdentityDenotTransformer {
     * before this phase starts processing same tree */
   override def runsAfter = Set(classOf[Mixin])
 
+  def isCompanionNeeded(cls: ClassSymbol)(implicit ctx: Context): Boolean = {
+    println("testing " + cls)
```
Comment (smarter) [inline src/dotty/tools/dotc/transform/LazyVals.scala]: I assume you'll drop the println before merging?

Comment (DarkDimius) [inline src/dotty/tools/dotc/transform/LazyVals.scala]: Thanks for noticing.

**Comment by smarter** [comment]
I think DropEmptyCompanions.scala should be deleted too, otherwise it's easy to think that it's still in use.

**Comment by odersky** [comment]
@smarter There are actually some other phases as well which are still in transfrom even though they are not used. We could get rid of them all maybe, or rename to .scala.disabled (?)

**Comment by DarkDimius** [comment]
rebased.

**Comment by odersky** [comment]
LGTM

**Pull request merged**